db/crud.go

```diff
@@ -1316,9 +1316,9 @@ func validateNewBody(body Body) error {
 		return base.HTTPErrorf(http.StatusNotFound, "Document revision is not accessible")
 	}
 
-	// Reject bodies containing user special properties for compatibility with CouchDB
-	if containsUserSpecialProperties(body) {
-		return base.HTTPErrorf(400, "user defined top level properties beginning with '_' are not allowed in document body")
+	// Reject bodies that contains the "_purged" property.
+	if body[BodyPurged] != nil {
```

Comment:
As I recall, _purged is just used by Sync Gateway to identify documents that shouldn't be imported (when we need to create/delete a document to trigger tombstone compaction from indexes).  I see that BodyPurged wasn't included in the set of allowed properties in containsUserSpecialProperties, but I'm not sure that was intentional.  Are you thinking that the omission from containsUserSpecialProperties was providing an accidental guard against BodyPurged that needs to be maintained?

Reply:
I did think that it was providing a guard against `BodyPurged` which is why I made the validation case. However since it was accidentally omitted from the `containsUserSpecialProperties` and it is only needed for import, I have removed the validation case. 
Import does read the `BodyPurged` and reject the document before it gets to the validation function.